snap_http/http.py

```diff
@@ -15,6 +16,13 @@
 class SnapdHttpException(Exception):
     """An exception raised during HTTP communication with snapd."""
 
+    @cached_property
+    def json(self):
+        """Attempts to parse the body of this exception as json."""
+        body = self.args[0]
```

Comment:
Generally speaking it's unlikely we'll raise a `SnapdHttpException` with no `args`, but since we don't make it impossible, it's probably worth being careful about getting an `IndexError` here.

Reply:
Handled that case